File: C++/1807. 【认证试题】矩阵转置.cpp
/*
 * Copyright (c) Huawei Technologies Co., Ltd. 2020-2020. All rights reserved.
 * Description: 上机编程认证
 * Note: 缺省代码仅供参考，可自行决定使用、修改或删除
 */
#include <iostream>
#include <sstream>
#include <string>
#include <cmath>
#include <utility>
#include <algorithm>
using namespace std;

class Solution {
public:
    // 待实现函数，在此函数中填入答题代码
    string Transpose(const string& str)
    {
        string temp = "";
        int n = sqrt(str.size() + 1);
        
        if(n * n != str.size()) {
            return "ERROR";
        }
        
        for(int i=0;i<n;++i) {
            for(int j=0;j<n * n;j = j + n) {
                temp += str[i + j];
            }
        }
        return temp;
    }
};

int main()
{
    string str;
    cin >> str;
    
    Solution solu;
    string output = solu.Transpose(str);
    cout << output << endl;

    return 0;
}

File: Python/1807. 【认证试题】矩阵转置.py
'''
 Copyright (c) Huawei Technologies Co., Ltd. 2019-2020. All rights reserved.
 Description: 考生实现代码
 Note: 缺省代码仅供参考，可自行决定使用、修改或删除
'''

import math
class Solution:
    def transpose(self, items):
        temp = ""
        n = int(math.sqrt(len(items)))
        if n * n != len(items):
            return "ERROR"
        
        for i in range(n):
            for j in range(0, n * n, n):
                temp += items[i + j]
        
        return temp


if __name__ == "__main__":
    items = input().strip()
    function = Solution()
    result = function.transpose(items)
    print(result)

